plugins/services/src/js/components/ServiceStatusIcon.tsx

```diff
@@ -104,35 +123,33 @@ class ServiceStatusIcon extends Component {
     );
   }
 
-  getUnableToLaunchWarning(service) {
-    if (this.isUnableToLaunch(service)) {
-      return this.getTooltip(
-        <Trans render="span">
-          There are tasks in this queue that DC/OS has failed to deploy for{" "}
-          {DateUtil.getDuration(
-            Date.now() - DateUtil.strToMs(service.getQueue().since)
-          )}
-          .
-        </Trans>
-      );
-    }
+  getUnableToLaunchWarning(service: Service) {
+    const duration = DateUtil.getDuration(
+      Date.now() - DateUtil.strToMs((service.getQueue() as any).since as string)
+    );
 
-    return null;
+    return this.getTooltip(
+      <Trans render="span">
+        There are tasks in this queue that DC/OS has failed to deploy for
+        {duration}.
+      </Trans>
+    );
   }
 
-  isUnableToLaunch(service) {
-    const queue = service.getQueue();
+  isUnableToLaunch(service: ServiceOrPodOrTree) {
+    const queue: any = service.getQueue();
```

Comment:
I think this is an array of objects.

Reply:
i don't have enough experience with the codebase to be capable of typing things easily. i think it's better to keep `any` here until we have a good type for `Queue`, as `any` is kind of screaming for attention while it can look like someone has already made a good decision in typing it (while `object[]` does not seem quite correct here).

so i'd propose to keep it `any` until we have introduced more typing to structs.

Reply:
Okay, but maybe add a todo?

Reply:
i anticipate us enforcing a `no-any` policy (tslint has a checker for that) in the future, so we need to get rid of all the `any`s left in the code. until then i think we need to find a balance of typing and being able to get work done. i'd consider having a todo at every `any` overkill (as there will be quite some and they can easily be found with the help of `tslint`).

what do you think?

Reply:
I don't mind `any`-s in some cases, because sometimes `any` really is the best we can do, but `queue` is something that we can easily verify and see the type of, so adding the type should be pretty easy to do, even if we don't want this type in this PR. I am not saying to add a todo to every any (just the ones that should be a quick fix) and `no-any` policy doesn't seem like something we can and will do in the near future.

Reply:
i am not sure about the type of queue. to me it looks like it is at least a 

```
type Queue = Array<{
  app: Application,
  count: number,
  delay: {
    timeLeftSeconds: number,
    overdue: boolean
  },
  since: string // iso8601,
  processedOfferSummary: OfferSummary[],
  //...

}>
```

i do not dare to add that, as i am missing too many concepts and knowledge here yet.

Reply:
Yeah, this is why we can add a todo, because it's an `any` that is not impossible to determine.

Reply:
i am a strong believer in that `no-any` is very possible, as i worked with one large TS-codebase without an `any` and know that every elm-codebase is completely typed without any being even a possibility. i've not come across a use case where `any` could not be replaced with proper types.

Reply:
It's possible if the project is started with typescript in mind, but currently we have over 200 `any`-s without counting the private repository. Even if it's possible, it's so much work that I doubt we will ever do it. I think this is a good sync topic, so I added it to our list.